server/test/durablejobs/StartupDurableJobRunnerTest.java

```diff
@@ -0,0 +1,213 @@
+                "it_email_address",
+                "test@example.com",
+                "base_url",
+                "https://civiform-test.dev",
+                "durable_jobs.job_timeout_minutes",
+                0,
+                "durable_jobs.poll_interval_seconds",
+                0));
+
+    durableJobRegistry = new DurableJobRegistry();
+
+    recurringDurableJobRunner =
+        new StartupDurableJobRunner(
+            config,
+            instanceOf(DurableJobExecutionContext.class),
+            durableJobRegistry,
+            instanceOf(PersistedDurableJobRepository.class),
+            () ->
+                instanceOf(
+                    new BindingKey<>(LocalDateTime.class)
+                        .qualifiedWith(BindingAnnotations.Now.class)),
+            simpleEmailMock,
+            instanceOf(ZoneId.class));
+  }
+
+  @Test
+  public void runJobs_timesOut() {
+    durableJobRegistry.registerWithNoTimeResolver(
+        DurableJobName.TEST,
+        JobType.RUN_ONCE,
+        (persistedDurableJob) ->
+            makeTestJob(
+                persistedDurableJob,
+                () -> {
+                  try {
+                    Thread.sleep(/* millis= */ 3000L);
+                  } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                  }
+                }));
+
+    PersistedDurableJobModel job = createPersistedJobToExecute();
+
+    recurringDurableJobRunner.runJobs();
+
+    job.refresh();
+    assertThat(job.getErrorMessage().get()).contains("JobRunner_JobTimeout");
+  }
+
+  @Test
+  public void rubJobs_executionException() {
+    durableJobRegistry.registerWithNoTimeResolver(
+        DurableJobName.TEST,
+        JobType.RUN_ONCE,
+        (persistedDurableJob) ->
+            makeTestJob(
+                persistedDurableJob,
+                () -> {
+                  throw new RuntimeException("test-execution-exception");
+                }));
+
+    PersistedDurableJobModel job = createPersistedJobToExecute();
+
+    recurringDurableJobRunner.runJobs();
+
+    job.refresh();
+
+    assertThat(job.getErrorMessage().get()).contains("JobRunner_JobFailed");
+    assertThat(job.getRemainingAttempts()).isEqualTo(0);
+  }
+
+  @Test
+  public void runJobs_runsJobsThatAreReady() {
+    AtomicInteger runCount = new AtomicInteger(0);
+    durableJobRegistry.registerWithNoTimeResolver(
+        DurableJobName.TEST,
+        JobType.RUN_ONCE,
+        (persistedDurableJob) ->
+            makeTestJob(persistedDurableJob, () -> runCount.getAndIncrement()));
+
+    PersistedDurableJobModel jobA = createPersistedJobToExecute();
+    PersistedDurableJobModel jobB = createPersistedJobToExecute();
+    PersistedDurableJobModel jobC = createPersistedJobScheduledInFuture();
+
+    recurringDurableJobRunner.runJobs();
+
+    jobA.refresh();
+    jobB.refresh();
+    jobC.refresh();
+
+    // This assertion fails occasionally. I've been unable to figure out why
```

Comment:
**question:** Does this comment still apply?

Reply:
I don't think so. I haven't had it fail at all.